<!-- Main panda Autograsp launchfile
    This launch file launches all the required ROS nodes except the panda_autograsp_cli.py command line interface (CLI).
    This CLI can be opened using the 'rosrun panda_autograsp panda_autograsp_cli.py' command.
-->
<launch>

    <!--Launch file arguments-->
    <arg name="base_name"           default="panda_autograsp"/>

    <!-- Start the Kinect processing ROS nodes
        The IAI_kinect2 package was used for the kinect processing.
    -->
    <include file="$(find panda_autograsp)/launch/kinect2_bridge.launch" >
        <arg name="output" value="log" />
    </include>

    <!--Create panda_autograsp group-->
    <group ns="$(arg base_name)">

        <!-- Load Rviz-->
        <include file="$(find panda_autograsp)/launch/panda_autograsp_rviz.launch">
            <arg name="output" value="log" />
        </include>

        <!-- Load dynamic reconfigure GUI -->
        <node pkg="rqt_gui" type="rqt_gui" name="dyn_reconf_gui" args="-s reconfigure" output="log"/>

        <!-- Create static TF for the calibration board (CB)
            The pose for this static TF can be set in the calib_frame_pose.yaml file.

            TF broadcaster conventions:
                yaw: is rotation about Z (blue axis)
                pitch: is rotation about Y (red axis)
                roll: is rotation about X (green axis)
        -->
        <rosparam command="load" file="$(find panda_autograsp)/cfg/calib_frame_pose.yaml" ns="tf2_broadcaster"/> <!-- Load calibration parametesr -->
        <node name="tf2_broadcaster" pkg="panda_autograsp" type="tf2_broadcaster.py" args="panda_link0 calib_frame" output="log"/>

        <!-- Load panda_autograsp services
            Panda_autograsp services:
               - panda_autograsp: Main panda_autograsp server. Used for calling all the other services.
               - moveit_planner_server: Used for controlling the robot.
               - grasp_planner_server: Used for computing the grasps.
        -->
        <node name="panda_autograsp_server" pkg="panda_autograsp" type="panda_autograsp_server.py" output="screen"/>
        <node name="grasp_planner" pkg="panda_autograsp" type="grasp_planner_server.py" output="log"/>
        <node name="moveit_planner_server" pkg="panda_autograsp" type="moveit_planner_server.py" output="screen">
            <remap from="execute_plan" to="moveit/execute_plan"/> <!--Remaped to make better distinction between grasp planner service-->
            <remap from="plan_to_point" to="moveit/plan_to_point"/> <!--Ditto-->
            <remap from="plan_to_path" to="moveit/plan_to_path"/> <!--Ditto-->
            <remap from="plan_random_pose" to="moveit/plan_random_pose"/> <!--Ditto-->
            <remap from="plan_random_joint" to="moveit/plan_random_joint"/> <!--Ditto-->
            <remap from="plan_random_path" to="moveit/plan_random_path"/> <!--Ditto-->
            <remap from="visualize_plan" to="moveit/visualize_plan"/> <!--Ditto-->
        </node>
        </group>
</launch>